### ISSUE TITLE
Add a button to synchronize x-axis position/zoom across graphs

### ISSUE THREAD

**Issue by leleogere** [tensorflow/tensorboard#6724]
Sometimes, I need to zoom on the x-axis to get a more precise idea of the training at a particular time. However, I would like to have an easy way to be able to sync this zoom across all my metrics. For example, if zooming on step 1200 on the loss graph, I would like to be able to get the corresponding step in the accuracy graph. For now, I need to zoom each graph independently and it's quite tedious to do.

Ideally, it would be nice to be able to scroll right on the graph, and have this scroll sync across the others in order to keep the same time frame on all graphs.

This would also be very useful when I have a lot a runs, and I want to use the "Wall" method for the x-axis to avoid having runs overlapping each other. This would allow me to go through runs by scrolling to the right, and keeping all graphs zoomed on the same runs. You can find an example of this issue below, I zoomed on a few runs in the loss graph, but the two other graphs are still zoomed out and hard to read.
![image](https://github.com/tensorflow/tensorboard/assets/71326140/9e67f8d9-e1c0-4d45-946d-96f3b224e7fd)

I do not know what the best UI/UX would be to allow such a behavior, maybe a simple toggle "Sync x-axis across graphs" would be enough?

**Comment by yatbear** [comment]
Hi @leleogere,

We do have a `Link by step` feature (you can find the checkbox in the right `Settings` pane in the Time Series dashboard) as shown in the gif below:

![link_by_step](https://github.com/tensorflow/tensorboard/assets/15273931/8b2f5d26-9dde-4413-b668-b1d700c41c23)

Note that since the steps might be downsampled (https://github.com/tensorflow/tensorboard/blob/master/README.md#is-my-data-being-downsampled-am-i-really-seeing-all-the-data), the "linked step" for each card/graph will be the closest step possible if the exact step is not included due to downsampling.

**Comment by leleogere** [comment]
Yes I do know this feature, but there are 2 issues with it:
- When you have a lot of steps in the x-axis, linking by step might not be enough to see some details in the plots, and it would be more convenient to be able to zoom (in all plot at the same time).
- This method does not work when the horizontal axis is set to "Wall" (as well as "Relative), the checkbox is disabled.

**Comment by yatbear** [comment]
Hi @leleogere,

Thanks for the feedback. Marking this as a feature request for our team to take a look when there is bandwidth.
